README.md

```diff
@@ -47,6 +49,14 @@ Make sure Docker is running on your machine and then:
 - Run the integration tests: `sbt it:test`
 - Run all the tests: `sbt testAll`
 
+#### Client side
+
+The client tests are run using jest. To run them use `yarn test`. To add new tests just add them to a `__tests__` directory with a `.spec.js` suffix (anything without this suffix is skipped and deemed to be a helper for the tests).
+
+Before running the integration tests the server needs to be running with access to all the client side dependences (i.e. webpack needs to be running unless there are built bundled js files available) and you need to be authed to access the app. To set your credentials for the test environment run `node ./client-integration/helpers/setCredential.js`. This will open a browser to log into. Once you're logged in just quit the browser and run the tests.
+
```

Comment:
Can we add the `setCredentials` script as a part of a task that run tests in `package.json` or does it have to be run as a separate command?

Reply:
We actually use a temporary `user_data_directory` for Chromium (the browser being run by puppeteer) so that when you quit the browser after setting credentials, the credentials are preserved for the test run.

This means you get to run tests without logging in for the whole lifetime of the cookie rather than logging in on each test.

We _could_ do a credentials "check" quickly before a test ... i.e. open a browser and see whether we're redirected to Google. Seems a bit hacky ... what do you think?

Reply:
@Reettaphant this seems like it may be slightly fiddlier than I thought ... because of the initital SSL cert error if you haven't logged in then you don't actually get redirected to Google so it looks like you're on production metrics. Also you can't programatically check for an unsigned SSL cert with puppeteer. So will card this.